src/fides/api/graph/config.py

```diff
@@ -437,6 +437,7 @@ class Collection(BaseModel):
     erase_after: Set[CollectionAddress] = set()
     # An optional set of dependent fields that need to be queried together
     grouped_inputs: Set[str] = set()
+    data_categories: Set[FidesKey] = set()
```

Comment:
One test I'd like to see (could be manual) is to understand the side effects of existing RequestTasks in the database for DSR 3.0 that don't have the `data_categories` field.

DSR 3.0 builds the graph up front and stores its nodes as RequestTasks in the db with a json representation of the collection with incoming and outgoing edges.  I'd want to double check what happens if you have RequestTasks that have already been created but haven't run yet when this release goes out.  

For example, you might run a PrivacyRequest with DSR 3.0 and force a failure on one of the nodes on main, then checkout this branch and reprocess that privacy request.  Does the absence of the `data_categories` key break anything?


Relevant place where it could fail: https://github.com/ethyca/fides/blob/dec991016e03840c919defff140510890959ac0e/src/fides/api/task/execute_request_tasks.py#L88

Reply:
Thanks for calling this out! Here are my manual testing steps:

1. Checkout main, hard-code `SQLConnector.retrieve_data` to raise an exception
2. Submit an access request, verify it fails
3. Checkout this feature branch, remove the hard-coded exception and reprocess the failed request
4. The request can be reprocessed successfully

I added a test to lock in this behavior
```
def test_parse_from_task_without_data_categories(self):
    """
    Verify that a collection stored without data categories can still be deserialized.
    """
    del serialized_collection["data_categories"]
    parsed = Collection.parse_from_request_task(serialized_collection)
    assert parsed.data_categories == set()
```

Reply:
Thanks for testing!